{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import tensorflow as tf\n",
    "import numpy as np\n",
    "import pickle\n",
    "import random\n",
    "import re\n",
    "\n",
    "from tensorflow.python.client import device_lib\n",
    "from datetime import date\n",
    "\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "import sys\n",
    "#sys.path.append('/export/sabrina/RNA_struc_pred/2020-09-16_discriminator')\n",
    "import rna_enc_pdb\n",
    "import discr_funcs as funcs\n",
    "import resnet_emb_funcs_elu_HPC_v2 as res_funcs\n",
    "\n",
    "import os\n",
    "os.environ[\"CUDA_DEVICE_ORDER\"] = \"PCI_BUS_ID\"\n",
    "os.environ[\"CUDA_VISIBLE_DEVICES\"] = '0'#'0,1'\n",
    "import pandas as pd\n",
    "column_seq   = 0\n",
    "column_z     = 1\n",
    "column_drift = 2\n",
    "\n",
    "\n",
    "def np_pad(matrix, pad_max, pad_val, n_classes, shift=0):\n",
    "    M_pad = np.full([pad_max, pad_max, n_classes], pad_val)\n",
    "    M_pad[shift:matrix.shape[0]+shift, shift:matrix.shape[1]+shift, :] = matrix\n",
    "    return M_pad\n",
    "\n",
    "\n",
    "params = {'seq_vocabsize': 4, 'N_value': 4, 'prob_N': 0.25, 'resnet_k': 3, 'data_format': 'channels_last', 'RNA_len':100,\n",
    "          'pre_activation': True, 'loss_filter_fn': False, 'weight_decay': 0.0001, 'optimizer_adam_beta1': 0.9, \n",
    "          'optimizer_adam_beta2': 0.997, 'optimizer_adam_epsilon': 1e-09, 'N_GPUS': 4, 'n_iters': 100000, \n",
    "          'use_outline_padding': True,\n",
    "          'default_batch_size': 100, \n",
    "          \n",
    "          'n_classes': 3, 'learning_rate': 0.0001, 'warmup_steps': 0, 'alpha': 0.01, \n",
    "          'lr_decay': 0.9988, 'use_regularization': True, 'RNA_length': 100, 'padding': 0, 'n_dense_filter': 100, \n",
    "          'data_augmentation_shift': False, 'atrous_rate': 4, 'use_dilation': True, 'dilation_stdev': 0.01, 'resnet_filter': 5, \n",
    "          'block_sizes': 8, \n",
    "          'drift_train_file': '/export/sabrina/RNA_struc_pred/2020-09-16_discriminator/data/drift_train_shape_100_07072020_hierclust293clust.pkl', 'drift_valid_file': '/export/sabrina/RNA_struc_pred/2020-09-16_discriminator/data/drift_valid_shape_100_07072020_hierclust293clust.pkl', 'mean_field_train_file': '/export/sabrina/RNA_struc_pred/2020-09-16_discriminator/data/mean_field_ckpt_07072020_complete_loss_big_lr1e6_no_shape/mean_field_train_L100_07072020_big_no_shape_492000iter.pkl', 'mean_field_valid_file': '/export/sabrina/RNA_struc_pred/2020-09-16_discriminator/data/mean_field_ckpt_07072020_complete_loss_big_lr1e6_no_shape/mean_field_valid_L100_07072020_big_no_shape_492000iter.pkl', 'struc_vocab': 700, 'ckpt_dir': '/export/sabrina/RNA_struc_pred/2020-09-16_discriminator/ckpt/discr_big_no_shape_07072020_trained20200916/', 'figs_dir': '/export/sabrina/RNA_struc_pred/2020-09-16_discriminator/ckpt/discr_big_no_shape_07072020_trained20200916//plots/'}\n",
    "params[\"N_GPUS\"] = 1\n",
    "params[\"default_batch_size\"] = 80\n",
    "params[\"batch_size\"] = params[\"default_batch_size\"] // params[\"N_GPUS\"]\n",
    "params.update({'dilation_stdev': 0.01, 'dilation_rate': 4} )\n",
    "batch_size = params['default_batch_size']\n",
    "\n",
    "\n",
    "enc_sq_pdb = rna_enc_pdb.RNAEncoder(L=params[\"RNA_length\"], args=\"seq\")\n",
    "enc_db_pdb = rna_enc_pdb.RNAEncoder(L=params[\"RNA_length\"], args=\"struc\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/gpfs/scratch/ramakers/rnafold/data/drifts_vqvae100_3d_8_160k_23022022.pkl', 'rb') as f:\n",
    "    traindata_df, valdata_df, testdata_df = pickle.load(f)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done.\n"
     ]
    }
   ],
   "source": [
    "with open('/gpfs/scratch/ramakers/rnafold/data/meanfields_23022022.pkl', 'rb') as f:\n",
    "    tmp_idx_dict, val_idx_dict, test_idx_dict = pickle.load(f)\n",
    "print('Done.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "traindata_df = traindata_df[['subseq', 'z']]\n",
    "valdata_df   = valdata_df[['subseq', 'z']]\n",
    "testdata_df  = testdata_df[['subseq', 'z']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "rnd_truth = []\n",
    "for s in traindata_df.index:\n",
    "    if 'seed' in s:rnd_truth.append(s)\n",
    "\n",
    "mf_keys = list(tmp_idx_dict.keys())\n",
    "trainvals = {}\n",
    "cnt = 0\n",
    "for s in rnd_truth:\n",
    "    pdb_key = s.split('.')[0][:6]\n",
    "    if pdb_key not in traindata_df.index:\n",
    "        cnt+=1\n",
    "    else:\n",
    "        pdb_key = s.split('.')[0][:6]\n",
    "        subseq = traindata_df.loc[pdb_key, 'subseq']\n",
    "        subseq = np.array(enc_sq_pdb.encode(subseq)).reshape(1,100)\n",
    "        z      = traindata_df.loc[pdb_key, 'z']\n",
    "        driftz = traindata_df.loc[s,'z']\n",
    "        trainvals.update({s : [subseq, z, driftz]})     \n",
    "\n",
    "for s in mf_keys:\n",
    "    pdb_key = s.split('.')[0][:6]\n",
    "    if pdb_key in traindata_df.index:\n",
    "        subseq = traindata_df.loc[pdb_key, 'subseq']\n",
    "        subseq = np.array(enc_sq_pdb.encode(subseq)).reshape(1,100)\n",
    "        z      = traindata_df.loc[pdb_key, 'z']\n",
    "        trainvals.update({s+'_mf' : [subseq, z, tmp_idx_dict[s][0]]}) \n",
    "        \n",
    "\n",
    "        \n",
    "rnd_truth = []\n",
    "for s in valdata_df.index:\n",
    "    if 'seed' in s:rnd_truth.append(s)\n",
    "        \n",
    "mf_keys = list(val_idx_dict.keys())\n",
    "validvals = {}\n",
    "cnt = 0\n",
    "for s in rnd_truth:\n",
    "    pdb_key = s.split('.')[0][:6]\n",
    "    if pdb_key not in valdata_df.index:\n",
    "        cnt+=1\n",
    "    else:\n",
    "        pdb_key = s.split('.')[0][:6]\n",
    "        subseq = valdata_df.loc[pdb_key, 'subseq']\n",
    "        subseq = np.array(enc_sq_pdb.encode(subseq)).reshape(1,100)\n",
    "        z      = valdata_df.loc[pdb_key, 'z']\n",
    "        \n",
    "        driftz = valdata_df.loc[s,'z']\n",
    "        validvals.update({s : [subseq, z, driftz]})     \n",
    "\n",
    "for s in mf_keys:\n",
    "    pdb_key = s.split('.')[0][:6]\n",
    "    if pdb_key in valdata_df.index:\n",
    "        subseq = valdata_df.loc[pdb_key, 'subseq']\n",
    "        subseq = np.array(enc_sq_pdb.encode(subseq)).reshape(1,100)\n",
    "        z      = valdata_df.loc[pdb_key, 'z']\n",
    "        validvals.update({s+'_mf' : [subseq, z, val_idx_dict[s][0]]}) \n",
    "        \n",
    "        \n",
    "        \n",
    "rnd_truth = []\n",
    "for s in testdata_df.index:\n",
    "    if 'seed' in s:rnd_truth.append(s)\n",
    "        \n",
    "mf_keys = list(test_idx_dict.keys())\n",
    "testvals = {}\n",
    "cnt = 0\n",
    "for s in rnd_truth:\n",
    "    pdb_key = s.split('.')[0][:6]\n",
    "    if pdb_key not in testdata_df.index:\n",
    "        cnt+=1\n",
    "    else:\n",
    "        pdb_key = s.split('.')[0][:6]\n",
    "        subseq = testdata_df.loc[pdb_key, 'subseq']\n",
    "        subseq = np.array(enc_sq_pdb.encode(subseq)).reshape(1,100)\n",
    "        z      = testdata_df.loc[pdb_key, 'z']\n",
    "        driftz = testdata_df.loc[s,'z']\n",
    "        testvals.update({s : [subseq, z, driftz]})     \n",
    "\n",
    "for s in mf_keys:\n",
    "    pdb_key = s.split('.')[0][:6]\n",
    "    if pdb_key in testdata_df.index:\n",
    "        subseq = testdata_df.loc[pdb_key, 'subseq']\n",
    "        subseq = np.array(enc_sq_pdb.encode(subseq)).reshape(1,100)\n",
    "        z      = testdata_df.loc[pdb_key, 'z']\n",
    "        testvals.update({s+'_mf' : [subseq, z, test_idx_dict[s][0]]}) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### sanity check for: shapes match\n",
    "for k in trainvals.keys():\n",
    "    k1 = trainvals[k][1]\n",
    "    k2 = trainvals[k][2]\n",
    "    \n",
    "    if k1.shape != k2.shape:\n",
    "        print(k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "trainvals = pd.DataFrame.from_dict(trainvals, orient='index')\n",
    "trainvals['pdbid'] = trainvals.index\n",
    "\n",
    "validvals = pd.DataFrame.from_dict(validvals, orient='index')\n",
    "validvals['pdbid'] = validvals.index\n",
    "\n",
    "testvals = pd.DataFrame.from_dict(testvals, orient='index')\n",
    "testvals['pdbid'] = testvals.index\n",
    "\n",
    "traindata = trainvals.values\n",
    "valdata = validvals.values\n",
    "testdata = testvals.values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# define model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def model_fn(inputs_raw, emb_raw, shift_vals, params, training):\n",
    "    \n",
    "    # onehot.shape == [b,L,4]\n",
    "    onehot = tf.one_hot(indices=inputs_raw, depth=params[\"seq_vocabsize\"])        \n",
    "    # check if \"N\" (== value of 4) is in inputs and mask every nucleotide in onehot with 0.25\n",
    "    equal_N = tf.to_float(tf.equal(inputs_raw, params[\"N_value\"])) * params[\"prob_N\"]\n",
    "    tile_eq_N = tf.tile(tf.expand_dims(equal_N, axis=-1), [1,1,params[\"seq_vocabsize\"]])\n",
    "    onehot = onehot + tile_eq_N\n",
    "    \n",
    "    # reshape_onehot.shape == [b,4,L,1], reshape_onehot_T.shaoe == [b,4,1,L]\n",
    "    reshape_onehot = tf.expand_dims(tf.transpose(onehot, perm=[0,2,1]), axis=3)\n",
    "    reshape_onehot_T = tf.transpose(reshape_onehot, perm=[0,1,3,2])\n",
    "    # s.shape == [b,4,L,L], sT.shape == [b,4,L,L]\n",
    "    s = reshape_onehot * tf.ones_like(reshape_onehot_T)\n",
    "    sT = reshape_onehot_T * tf.ones_like(reshape_onehot)\n",
    "    # M.shape == [b,8,L,L]\n",
    "    M = tf.concat([s,sT], axis=1)\n",
    "    \n",
    "    # pad original seq_len to max_len with pad_val == -1\n",
    "    equal = tf.to_float(tf.not_equal(inputs_raw, -1))\n",
    "    tile = tf.tile(tf.reshape(equal, [-1, 1, params[\"RNA_length\"], 1]), [1, 1, 1, params[\"RNA_length\"]])\n",
    "    pad_mask = tile * tf.transpose(tile, perm=[0,1,3,2]) \n",
    "    M_pad = M * pad_mask\n",
    "    \n",
    "    # onehot_tar.shape == [b,n_classes,L,L]\n",
    "    onehot_emb = tf.to_float(tf.transpose(emb_raw, perm=[0,3,1,2]))\n",
    "    \n",
    "    # diag frame.shape == [b,1,L,L]\n",
    "    diag_frame = tf.matrix_band_part(tf.ones([tf.shape(M_pad)[0], 1, params[\"RNA_length\"], params[\"RNA_length\"]]), 0, 0)\n",
    "    diag_frame = diag_frame * pad_mask\n",
    "    \n",
    "    # t_STACK.shape == [b,L,L,8+1+n_classes]\n",
    "    STACK = tf.concat([M_pad, onehot_emb, diag_frame], axis=1)\n",
    "    t_STACK = tf.transpose(STACK, perm=[0,2,3,1])\n",
    "    print(\"t_STACK.shape ==\", t_STACK.shape)\n",
    "    \n",
    "    \n",
    "    inputs = res_funcs.conv2d_layer(t_STACK, params, 'conv_prev', args='first_conv', raw_sq_input=inputs_raw, pad_mask=pad_mask, shift_vals=shift_vals)\n",
    "    #return inputs\n",
    "    \n",
    "    for i, num_blocks in enumerate(range(params[\"block_sizes\"])):\n",
    "        inputs = res_funcs.block_layer(i, inputs, training, params)\n",
    "    \n",
    "    if params[\"pre_activation\"]:\n",
    "        inputs_bn   = res_funcs.batch_norm(inputs, training, params[\"data_format\"], 'batch_norm_pre_activation')\n",
    "        inputs_relu = tf.nn.relu(inputs_bn)\n",
    "        dense1 = tf.layers.conv2d(inputs=inputs_relu, filters=params[\"n_dense_filter\"], kernel_size=[1,1], strides=[1,1], padding='VALID', name='dense1')\n",
    "    else:\n",
    "        inputs_relu = tf.nn.relu(inputs)\n",
    "        dense1 = tf.layers.conv2d(inputs=inputs_relu, filters=params[\"n_dense_filter\"], kernel_size=[1,1], strides=[1,1], padding='VALID', name='dense1')\n",
    "    \n",
    "    dense2 = tf.nn.relu(dense1)\n",
    "    dense2 = tf.layers.conv2d(inputs=dense2, filters=1, kernel_size=[1,1], strides=[1,1], padding='VALID', name='dense2')\n",
    "    \n",
    "    # logits.shape == [b,1]\n",
    "    logits = tf.reduce_sum(dense2, axis=[1,2])\n",
    "    \n",
    "    return logits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "80"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "params[\"batch_size\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "device_list: ['/device:GPU:0'] \n",
      "\n",
      "t_STACK.shape == (?, 100, 100, 12)\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162acae80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162acae80>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162acae80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162acae80>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ac62b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ac62b0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ac62b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ac62b0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162afea20>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162afea20>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162afea20>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162afea20>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ad1ef0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ad1ef0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ad1ef0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162ad1ef0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162aef198>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162aef198>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162aef198>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162aef198>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162b37828>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162b37828>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162b37828>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162b37828>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162bff208>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162bff208>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162bff208>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162bff208>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162c792e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162c792e8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162c792e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162c792e8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162d559b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162d559b0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162d559b0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162d559b0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162e00e80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162e00e80>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162e00e80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162e00e80>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162de0b38>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162de0b38>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162de0b38>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162de0b38>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162c93d68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162c93d68>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162c93d68>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab162c93d68>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f224e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f224e0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f224e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f224e0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f4b940>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f4b940>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f4b940>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162f4b940>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1630efb70>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1630efb70>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1630efb70>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1630efb70>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162df1160>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162df1160>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162df1160>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab162df1160>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1630ef438>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1630ef438>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1630ef438>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1630ef438>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16317c2e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16317c2e8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16317c2e8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16317c2e8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16316b550>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16316b550>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16316b550>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16316b550>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1632c9a20>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1632c9a20>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1632c9a20>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1632c9a20>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1633a4080>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1633a4080>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1633a4080>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1633a4080>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16317c748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16317c748>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16317c748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16317c748>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16345c048>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16345c048>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16345c048>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16345c048>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163506390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163506390>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163506390>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163506390>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163439400>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163439400>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163439400>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163439400>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163398518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163398518>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163398518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab163398518>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab163398518>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "t_STACK.shape == (?, 100, 100, 12)\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1668eb6d8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1668eb6d8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1668eb6d8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1668eb6d8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668efe80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668efe80>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668efe80>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668efe80>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1669144e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1669144e0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1669144e0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab1669144e0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668f8588>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668f8588>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668f8588>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1668f8588>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166945198>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166945198>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166945198>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166945198>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16694ec50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16694ec50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16694ec50>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab16694ec50>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1669550f0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1669550f0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1669550f0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab1669550f0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16691fbe0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16691fbe0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16691fbe0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab16691fbe0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166a0d048>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166a0d048>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166a0d048>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166a0d048>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ac7cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ac7cf8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ac7cf8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ac7cf8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166afe320>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166afe320>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166afe320>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166afe320>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166b8ccc0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166b8ccc0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166b8ccc0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166b8ccc0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166bbaf60>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166bbaf60>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166bbaf60>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166bbaf60>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166af6710>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166af6710>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166af6710>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166af6710>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166cc31d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166cc31d0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166cc31d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166cc31d0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166b64860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166b64860>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166b64860>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166b64860>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166cc36d8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166cc36d8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166cc36d8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166cc36d8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166d86978>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166d86978>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166d86978>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166d86978>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d307b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d307b8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d307b8>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d307b8>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d5df98>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d5df98>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d5df98>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166d5df98>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166e93780>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166e93780>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166e93780>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166e93780>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e5db70>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e5db70>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e5db70>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e5db70>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e93748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e93748>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e93748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166e93748>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166f048d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166f048d0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166f048d0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166f048d0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166f04fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166f04fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166f04fd0>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166f04fd0>>: AssertionError: Bad argument number for Name: 3, expecting 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ee8e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ee8e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ee8e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method BatchNormalization.call of <tensorflow.python.layers.normalization.BatchNormalization object at 0x2ab166ee8e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING:tensorflow:Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "WARNING: Entity <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Conv.call of <tensorflow.python.layers.convolutional.Conv2D object at 0x2ab166ee8e10>>: AssertionError: Bad argument number for Name: 3, expecting 4\n",
      "logits_1.shape == (80, 1)\n",
      "logits_2.shape == (80, 1)\n",
      "\n",
      "Number of discr_params (energy network): 5056 \n",
      "\n",
      "\n",
      "Done graph init...\n"
     ]
    }
   ],
   "source": [
    "tf.reset_default_graph()\n",
    "g = tf.Graph()\n",
    "with g.as_default():\n",
    "    \n",
    "    with tf.variable_scope('resnet', reuse=tf.AUTO_REUSE):\n",
    "        \n",
    "        tconfig=tf.ConfigProto(allow_soft_placement=True, log_device_placement=True)\n",
    "        c1 = []\n",
    "        c2 = []\n",
    "        device_list = [x.name for x in device_lib.list_local_devices() if x.device_type == 'GPU']\n",
    "        print(\"device_list: %s \\n\" % device_list)\n",
    "        \n",
    "        sq_ph        = tf.placeholder(tf.int32, [None, params[\"RNA_len\"]])\n",
    "        emb_pdb_ph   = tf.placeholder(tf.int32, [None, params[\"RNA_len\"], params[\"RNA_len\"]])\n",
    "        emb_drift_ph = tf.placeholder(tf.int32, [None, params[\"RNA_len\"], params[\"RNA_len\"]])\n",
    "        train_ph     = tf.placeholder(tf.bool, ())\n",
    "        cl_ph        = tf.placeholder(tf.float32, [None, 1])\n",
    "        shift_ph     = tf.placeholder(tf.int32, [None,1])\n",
    "        step_ph      = tf.placeholder(tf.float32, ())\n",
    "        \n",
    "\n",
    "        \n",
    "        start_idx = 0\n",
    "        #if True:\n",
    "        for d in device_list:\n",
    "            #if True:\n",
    "            with tf.device(d):\n",
    "                #print(\"\\nwith device %s \\n\" % d)\n",
    "                sq_gpu        = sq_ph[start_idx:start_idx + params[\"batch_size\"]]\n",
    "                emb_pdb_gpu   = tf.one_hot(emb_pdb_ph[start_idx:start_idx + params[\"batch_size\"]], params[\"n_classes\"] )\n",
    "                emb_drift_gpu = tf.one_hot(emb_drift_ph[start_idx:start_idx + params[\"batch_size\"]], params[\"n_classes\"] )\n",
    "                cl_gpu        = cl_ph[start_idx:start_idx + params[\"batch_size\"]]\n",
    "                shift_gpu     = shift_ph[start_idx:start_idx + params[\"batch_size\"]]\n",
    "                \n",
    "                \n",
    "                #ym_mask = res_funcs.get_target_mask_distributed_v2(sq_gpu, shift_gpu, discr_params)\n",
    "                #nd_mask = sym_mask #* tf.random.uniform((discr_params['batch_size'], params[\"n_classes\"], params[\"RNA_length\"], params[\"RNA_length\"]))\n",
    "                \n",
    "                logits_1 = model_fn(sq_gpu, emb_pdb_gpu, shift_gpu, params,train_ph)\n",
    "                logits_2 = model_fn(sq_gpu, emb_drift_gpu, shift_gpu, params, train_ph)\n",
    "                logits = logits_1 - logits_2\n",
    "                print('logits_1.shape == %s' % logits_1.shape)\n",
    "                print('logits_2.shape == %s' % logits_2.shape)\n",
    "                print(\"\\nNumber of discr_params (energy network): %s \\n\" % funcs.get_num_of_params())\n",
    "                \n",
    "                loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(labels=cl_gpu, logits=logits))\n",
    "                c1.append(loss)\n",
    "                \n",
    "                preds = tf.round(tf.nn.sigmoid(logits))\n",
    "                acc = tf.reduce_mean(tf.cast(tf.equal(preds, cl_gpu), tf.float32))\n",
    "                c2.append(acc)\n",
    "                \n",
    "            start_idx = start_idx + params[\"batch_size\"]\n",
    "                \n",
    "        with tf.device('/cpu:' + str([x.name for x in device_lib.list_local_devices() if x.device_type == 'CPU'][0][-1])):\n",
    "            loss = tf.add_n(c1) / params['N_GPUS']\n",
    "            acc  = tf.add_n(c2) / params['N_GPUS']\n",
    "            \n",
    "            \n",
    "            def exclude_batch_norm(name):\n",
    "                return 'batch_normalization' not in name\n",
    "\n",
    "            if params[\"use_regularization\"]:\n",
    "                loss_filter_fn = params[\"loss_filter_fn\"] or exclude_batch_norm\n",
    "                l2_loss = params[\"weight_decay\"] * tf.add_n([tf.nn.l2_loss(tf.cast(v, tf.float32)) for v in tf.trainable_variables() if loss_filter_fn(v.name)])\n",
    "                loss = loss + l2_loss\n",
    "            \n",
    "            \n",
    "            lr = funcs.get_lr(params[\"learning_rate\"], step_ph, params[\"lr_decay\"])\n",
    "            #lr = tf.constant(discr_params[\"learning_rate\"], dtype=tf.float32)\n",
    "            #lr = funcs.lr_decay(discr_params[\"learning_rate\"], params[\"lr_decay\"], step_ph)\n",
    "            optimizer = tf.contrib.opt.LazyAdamOptimizer(lr, beta1=params[\"optimizer_adam_beta1\"],\n",
    "                                                         beta2=params[\"optimizer_adam_beta2\"], \n",
    "                                                         epsilon=params[\"optimizer_adam_epsilon\"])\n",
    "            global_step = tf.train.get_global_step()\n",
    "            tvars = tf.trainable_variables()\n",
    "\n",
    "            gradients = optimizer.compute_gradients(loss, tvars, colocate_gradients_with_ops=True)\n",
    "            minimize_op = optimizer.apply_gradients(gradients, global_step=global_step, name=\"train\")\n",
    "            update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)\n",
    "            train_op = tf.group(minimize_op, update_ops)\n",
    "\n",
    "            init_op = tf.global_variables_initializer()\n",
    "            \n",
    "            \n",
    "        saver = tf.train.Saver()\n",
    "        \n",
    "        #print(\"\\nTRAINABLE VARIABLES:\")\n",
    "        #for i in tf.trainable_variables():\n",
    "        #    print(i)\n",
    "            \n",
    "    print(\"\\nDone graph init...\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "zero_shift    = np.array([[0] for _ in range(batch_size)])\n",
    "perfect_class = np.array([[1] for _ in range(batch_size)])\n",
    "\n",
    "def batch():\n",
    "    batch_rnd = traindata[np.random.choice(traindata.shape[0], batch_size, replace=False)]\n",
    "    pdb_ids = batch_rnd[:, -1]\n",
    "    return np.concatenate(batch_rnd[:,column_seq]), np.stack(batch_rnd[:,column_z]),\\\n",
    "            np.stack(batch_rnd[:,column_drift]),  pdb_ids \n",
    "\n",
    "\n",
    "def valbatch():\n",
    "    batch_rnd = valdata[np.random.choice(valdata.shape[0], batch_size, replace=True)]\n",
    "    pdb_ids = batch_rnd[:, -1]\n",
    "    return np.concatenate(batch_rnd[:,column_seq]), np.stack(batch_rnd[:,column_z]),\\\n",
    "            np.stack(batch_rnd[:,column_drift]),  pdb_ids "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done preprocessing.\n"
     ]
    }
   ],
   "source": [
    "print('Done preprocessing.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 10.064486 - "
     ]
    }
   ],
   "source": [
    "params[\"n_iters\"] = 1000000\n",
    "tlr_list = []\n",
    "tloss_list = []\n",
    "tacc_list = []\n",
    "vloss_list = []\n",
    "vacc_list = []\n",
    "with tf.Session(graph=g, config=tconfig) as sess:\n",
    "    sess.run(init_op)\n",
    "    \n",
    "    loss_lst, acc_lst, lr_lst = [], [], []\n",
    "    val_loss_lst, val_acc_lst, val_lr_lst, val_iter = [], [], [], []\n",
    "\n",
    "\n",
    "    for i in range(params[\"n_iters\"]):\n",
    "        \n",
    "        sq_batch, emb_pdb_batch, emb_drift_batch, pdbs = batch()\n",
    "        \n",
    "        \n",
    "        ###### random flipping of embedding vs drift ########\n",
    "        cls_labels = []\n",
    "        cnt = 0\n",
    "        for i1, i2 in zip( emb_pdb_batch, emb_drift_batch):\n",
    "            if np.random.rand() < 0.5:\n",
    "                cls_labels.append([0])\n",
    "                tmp = emb_pdb_batch[cnt].copy()\n",
    "                emb_pdb_batch[cnt]   = emb_drift_batch[cnt]\n",
    "                emb_drift_batch[cnt] = tmp\n",
    "            else:\n",
    "                cls_labels.append([1])\n",
    "            cnt += 1\n",
    "\n",
    "        cls_labels = np.array(cls_labels)\n",
    "\n",
    "\n",
    "        _, loss_np, acc_np, lr_np = sess.run([train_op, loss, acc, lr], feed_dict={sq_ph: sq_batch, \n",
    "                                                                                   emb_pdb_ph: emb_pdb_batch, \n",
    "                                                                                   emb_drift_ph: emb_drift_batch,\n",
    "                                                                                   cl_ph:     cls_labels  ,\n",
    "                                                                                   shift_ph: zero_shift,\n",
    "                                                                                   train_ph: True, step_ph: i})\n",
    "        tloss_list.append(loss_np)\n",
    "        tacc_list.append(acc_np)\n",
    "        tlr_list.append(lr_np)\n",
    "        \n",
    "        if (i % 1000) == 0:\n",
    "            \n",
    "            sq_batch, emb_pdb_batch, emb_drift_batch, pdbs = valbatch()\n",
    "        \n",
    "        \n",
    "            ###### random flipping of embedding vs drift ########\n",
    "            cls_labels = []\n",
    "            cnt = 0\n",
    "            for i1, i2 in zip( emb_pdb_batch, emb_drift_batch):\n",
    "                if np.random.rand() < 0.5:\n",
    "                    cls_labels.append([0])\n",
    "                    tmp = emb_pdb_batch[cnt].copy()\n",
    "                    emb_pdb_batch[cnt]   = emb_drift_batch[cnt]\n",
    "                    emb_drift_batch[cnt] = tmp\n",
    "                else:\n",
    "                    cls_labels.append([1])\n",
    "                cnt += 1\n",
    "\n",
    "            cls_labels = np.array(cls_labels)\n",
    "\n",
    "\n",
    "            vloss_np, vacc_np = sess.run([ loss, acc], feed_dict={sq_ph: sq_batch, \n",
    "                                                                                       emb_pdb_ph: emb_pdb_batch, \n",
    "                                                                                       emb_drift_ph: emb_drift_batch,\n",
    "                                                                                       cl_ph:     cls_labels  ,\n",
    "                                                                                       shift_ph: zero_shift,\n",
    "                                                                                       train_ph: True, step_ph: i})\n",
    "\n",
    "            vloss_list.append(vloss_np)\n",
    "            vacc_list.append(vacc_np)    \n",
    "            print(i, loss_np, end=' - ')\n",
    "            \n",
    "        if (i % 10000) == 0:\n",
    "            save_file =  '/gpfs/scratch/ramakers/discr/ckpt/discr_{0}.ckpt'.format(i)\n",
    "            save_path = saver.save(sess, save_file)\n",
    "            #print('##### saved ckpt:', i)\n",
    "            #plt.plot(range(len(loss_lst)), loss_lst),plt.show()\n",
    "            #print('~~~~~~~~~~~')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "#with open('/gpfs/scratch/ramakers/discr/ckpt/losses.pkl', 'wb') as f:\n",
    "#    pickle.dump([tloss_list, tacc_list, tlr_list, vloss_list, vacc_list], f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAGPxJREFUeJzt3Xt0HOWZ5/HvY8k32fK9bXyByDY24GQIEIWY2EOGQDJOIMPsLucsPpMMYZjjyewmy+xcsmaTJezubCADy2Rnk0niBIcMcJyEhAQmxmMcX2AAX5CxjW+yLduSLVm2WpJ1v0vP/tFlryxbaqm7dani9zmnj6rfervep0uln0qlqi5zd0REJPxGDXcBIiKSGQp0EZGIUKCLiESEAl1EJCIU6CIiEaFAFxGJCAW6iEhEKNBFRCIiaaCb2VozqzCzAz3av2JmhWZ20Mz+bvBKFBGR/sjuR59nge8A/3ShwczuAO4FPuzurWY2sz+DzZgxw/Py8lIoU0Tk/Wv37t2V7h5L1i9poLv7G2aW16P5z4En3L016FPRn6Ly8vIoKCjoT1cREQmYWUl/+qV6DH0x8LtmttPMXjezj/ZRyCozKzCzgng8nuJwIiKSTKqBng1MA5YCfwP83MzsSh3dfY2757t7fiyW9C8GERFJUaqBXgq85Am7gC5gRubKEhGRgUo10H8N3AFgZouBMUBlpooSEZGBS/pPUTNbB/weMMPMSoFvAGuBtcGpjG3AA64PVhcRGVb9OctlZS+zPp/hWkREJA26UlREJCJCEeinqpr412M65VFEpC/9uVJ02N3+5FYAip+4e5grEREZuUKxhy4iIskp0EVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCIU6CIiEaFAFxGJCAW6iEhEKNBFRCJCgS4iEhEKdBGRiFCgi4hEhAJdRCQikga6ma01s4rgdnM95/2VmbmZ6QbRIiLDrD976M8CK3o2mtnVwKeBUxmuSUREUpA00N39DaD6CrP+HvgqoJtDi4iMACkdQzeze4Eyd9/Xj76rzKzAzAricd1GTkRksAw40M0sB/ivwKP96e/ua9w9393zY7HYQIcTEZF+SmUPfSEwH9hnZsXAPOBdM7sqk4WJiMjADPgm0e6+H5h54XkQ6vnuXpnBukREZID6c9riOmA7cJ2ZlZrZQ4NfloiIDFTSPXR3X5lkfl7GqhERkZTpSlERkYhQoIuIRIQCXUQkIhToIiIRoUAXEYkIBbqISEQo0EVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCIU6CIiEaFAFxGJCAW6iEhEKNBFRCJCgS4iEhH9uWPRWjOrMLMD3dqeNLNCM3vPzH5lZlMGt0wREUmmP3vozwIrerRtAj7k7jcCR4FHMlyXiIgMUNJAd/c3gOoeba+5e0fwdAcwbxBqExGRAcjEMfQ/ATZkYDkiIpKGtALdzL4GdAAv9NFnlZkVmFlBPB5PZzgREelDyoFuZl8E7gH+yN29t37uvsbd8909PxaLpTqciIgkkZ3Ki8xsBfBV4BPu3pTZkkREJBX9OW1xHbAduM7MSs3sIeA7QC6wycz2mtn3B7lOERFJIukeuruvvELzM4NQi4iIpEFXioqIRIQCXUQkIhToIiIRoUAXEYkIBbqISEQo0EVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCIU6CIiEaFAFxGJCAW6iEhEKNBFRCJCgS4iEhEKdBGRiFCgi4hERH9uQbfWzCrM7EC3tmlmtsnMjgVfpw5umSIikkx/9tCfBVb0aFsNbHb3RcDm4LmIiAyjpIHu7m8A1T2a7wV+Ekz/BPjDDNclIiIDlOox9FnuXh5MnwVmZageERFJUdr/FHV3B7y3+Wa2yswKzKwgHo+nNMZDy+czcWx2qiWKiLwvpBro58xsNkDwtaK3ju6+xt3z3T0/FoulOJyIiCSTaqC/AjwQTD8AvJyZcnqX+ENARER605/TFtcB24HrzKzUzB4CngA+ZWbHgLuC54PGBnPhIiIRkfTAtLuv7GXWnRmuRURE0qArRUVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCIU6CIiERGaQNdlRSIifQtFoJuuLBIRSSoUgS4iIskp0EVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCIU6CIiERGaQNcNi0RE+haKQDddWSQiklRagW5m/9nMDprZATNbZ2bjMlWYiIgMTMqBbmZzgf8E5Lv7h4As4P5MFSYiIgOT7iGXbGC8mWUDOcCZ9EsSEZFUpBzo7l4GPAWcAsqBWnd/LVOFiYjIwKRzyGUqcC8wH5gDTDCzz1+h3yozKzCzgng8nnqlIiLSp3QOudwFnHT3uLu3Ay8BH+/Zyd3XuHu+u+fHYrE0hhMRkb6kE+ingKVmlmOJ8wrvBA5npiwRERmodI6h7wR+AbwL7A+WtSZDdV0+nu5ZJCLSp+x0Xuzu3wC+kaFaeqXLikREkgvFlaIiIpKcAl1EJCIU6CIiEaFAFxGJCAW6iEhEKNBFRCJCgS4iEhEKdBGRiAhNoOsWdCIifQtHoOtSURGRpMIR6CIikpQCXUQkIhToIiIRoUAXEYkIBbqISEQo0EVEIkKBLiISEWkFuplNMbNfmFmhmR02s9syVVhPuq5IRKRvad2CDvg/wL+4+31mNgbIyUBNlzFdWSQiklTKgW5mk4HbgS8CuHsb0JaZskREZKDSOeQyH4gDPzazPWb2IzObkKG6RERkgNIJ9GzgFuB77n4z0Ais7tnJzFaZWYGZFcTj8TSGExGRvqQT6KVAqbvvDJ7/gkTAX8Ld17h7vrvnx2KxNIYTEZG+pBzo7n4WOG1m1wVNdwKHMlKViIgMWLpnuXwFeCE4w+UE8GD6JYmISCrSCnR33wvkZ6gWERFJQ3iuFNWVRSIifQpFoJuuKxIRSSoUgS4iIskp0EVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCIU6CIiERGaQHddWSQi0qdQBLquKxIRSS4UgS4iIskp0EVEIkKBLiISEQp0EZGIUKCLiESEAl1EJCLSDnQzyzKzPWb2m0wUJCIiqcnEHvrDwOEMLKdPruuKRET6lFagm9k84G7gR5kpp7dxBnPpIiLRkO4e+reBrwJdGahFRETSkHKgm9k9QIW7707Sb5WZFZhZQTweT3U4ERFJIp099GXAH5hZMfBT4JNm9nzPTu6+xt3z3T0/FoulMZyIiPQl5UB390fcfZ675wH3A1vc/fMZq0xERAZE56GLiEREdiYW4u7bgG2ZWJaIiKRGe+giIhERmkDXdUUiIn0LRaCb7lkkIpJUKAJdRESSU6CLiESEAl1EJCIU6CIiEaFAFxGJCAW6iEhEKNBFRCIiNIHuumWRiEifQhHoumORiEhyoQh0ERFJToEuIhIRCnQRkYhQoIuIRIQCXUQkIlIOdDO72sy2mtkhMztoZg9nsjARERmYdG5B1wH8lbu/a2a5wG4z2+TuhzJUm4iIDEDKe+juXu7u7wbT9cBhYG6mCrtsvMFasIhIRGTkGLqZ5QE3AzuvMG+VmRWYWUE8Hk9t+WlVJyLy/pB2oJvZROCXwF+4e13P+e6+xt3z3T0/FoulO5yIiPQirUA3s9EkwvwFd38pMyWJiEgq0jnLxYBngMPu/nTmShIRkVSks4e+DPgC8Ekz2xs8Ppuhui4Rb2hDH7YoItK3lE9bdPc3GaL/V67bdQqA2qZ2JueMHoohRURCJ1RXirZ2dg53CSIiI1aoAr2ra7grEBEZuUIV6J06kC4i0qtwBXqnAl1EpDfhCnTtoYuI9Cpcgd6lQBcR6U2oAr2+pX24SxARGbFCFejP7zg13CWIiIxYoQp0ERHpXagC/Zfvlg53CSIiI1aoAl1ERHoXukBvauug8OxlH7vebxV1Lfgwnv54qqrpiu1HztZzvrGN4srGIa4oczq7nIeefYftx6s4W9sy4Nc3t3VSXts8CJVlRlNbBz94/bjOtuphKL5n5+pa+tym4vWtnKkZudvOUAldoC95dCMrvv2vtLR3crKykXh9K4++fICPP76ZE/EG8lavZ2thxWWvc3e2FlZw6zc388ybJ/nmq4f5298cuiTcWzs6OXimlraOLhpaOy55/Yl4AxV1iQ2q8GwdeavX871tx3F31u06xWOvHOSpjUcAeGJDIXmr19PV5ew9XXNxQ1z/Xjm3P7mVjQfPUlbTzM/fOU3e6vU88tJ+fv/bb3Dz/9zE7z21jdPVTTS3dfLkxkJaOzppbuu8+B5OVzdR3dh2sa62ji5qmtp4fkcJnV3O73xjI09uLGTdrlO8fbySl/eW8er+cprbEuvrZ++corqxjQ37yy9bR51dzg/fOMGuk9XUtbRzurqJ724t4rntxRw7V8/JykbqW9opq2lmz6nzVNS1kLd6PS/sLAGgor6FzYUVrPzhDpY+vpn6lnZKqhqpami9ZJyOzi6OnqvnF7tL6epyapvbeXlvGXc9/Tq3Pb6Fv3lxH3c8tY1zwfp2d94uqrz4vLva5naKKhLfm+a2Ts7UNHM83sD/3XyMp187wl1Pv85zO0po7PH9LOv2w1/X0k5rx+WfE/RWUeXFbam2qZ0lj27k8Q2FF99vY2sH54PvxRee2Une6vUXX9vV5ZdsW6eqmshbvZ7l39py2dlabxVV0tbRRXVjG20dXRff13M7SnB3mts62V1STWeXc7i8DvdLl11e23zJ8xPxBrq6nPdKa/jnfWeoamjlXF0LDa0dHD1Xf9n7/PWeMvJWr79iIL59vJLnthcD8POC07y8t4zXDp69OP/V/eXc9vgW3jxWyZGz9Ty/o4SqhlaOnK3nTE0zZ2qaae+89DM7LsyvamjlhZ0lF9dHc1snv9pTirvzTnE1TW3//3v2sW9uZunjmy++v+u+voE/e66AzYfPcb6xjY/+r9/y8Se28K1/KaSxtePimF3BOuu+LEicMXe+sY1/3FbEmZpmiirqWbfrFB3dam3v7OLzP9rJcztKePatkzS3dRKvb2XZE1v47aFzlNU0s+tkNT8NPjywq8upaWrD3dldcp6X3i3ldPWVd+AGiw3l3mp+fr4XFBQM+HXdf1D6a0z2qIs/HKn60icWMn3CGL7926M0tqX3wWCjDPq7Y/dfVlzP/rIaXt1/NnlnGVE+Nn8af/mpxfz7NTuGZLwpOaOpaRqe03mXXzuDN4sq+91/as5ozvej1qsmjePsFX55j2TXX5VL4dnLf1lesOza6bzwp0tTXr6Z7Xb3/KT9ohroIiIjyWOfW8IXl81P6bX9DfTQHXIREQmj3adqBn2MUAT6gtiE4S5BRCQt24/3//BUqtK9SfQKMztiZkVmtjpTRfX03+5eMliLFhEZEotn5Q76GOncJDoL+C7wGWAJsNLMBiV577h+5mAsVkRkyPyPez846GOkfE9R4FagyN1PAJjZT4F7gUOZKKyn4ifupq6lnfaOLsprW6hrbqco3sD8GRPIHTeaqoZWrp89iXdOVjNr0jh2naxmQWwCf7/pKAAPLstjSs4YvrJuDy9+6TZyx2VTXtvCnpLzzJ06nsWzcnnslYOcqW3h63ffwLypOUzJGU1LeyfuUN/SwaTx2ZyINzJ36njeLTnPpkPnWDRrIh+eN4WapnYKz9Zz701zqGluZ2rOaE5XN/PKvjKuv2oSd1w/k50nqpg2YQwvFpTyR0uv4YNzJvFWURWrbl/AhgPlNLR0MDlnDB/Nm4o7HDxTR2tHJ0+/dpRFsyZy++IY1Q1trN9fzp99YgHXxnKZN3U8xVWNPLejhDeOxlkyZzJLZk9i3a5T/OWnFpM7LpsPTM+ho9NpDc76Kapo4GRlI53u/Nub5zJj4lh+taeMD82dTFNbBycrG6lsaOWeG+dgwKhRRtn5ZpYvmsGp6ia2FVaw+Kpc5k4ZT5eDGUwaNxpwnthQyD03zqG4qpG5U8Yze/J49p4+z7JrZ3CurpXG1g5mThqLYTS3d3K+qY1brpnK/rIabp0/nZ+8XUxjawe/uyjGTVdP4anXjjAmexQ3XT2FmbljWRibSEV9C3nTJ/Djt4oxg/Fjsrj/o9fwnS1FFJRUc8+NsxmdNYqJY7OZkjOGoooGxo0eRWtHF1UNrSxfFON0dRNmMHFsNm8eq+TBZfN5dX85TW2d3HvTHDYfPsfSBdMpq2nmmmk5bD9Rxa3zp7HrZDX/tL2Ejy+czr/7yDxKzzdzw1W5nKxs5J3iau77yNW8sLOEnDFZPHznYjYdPscHpuXw7NvFPLgsj5KqJhbGJuI4V0/N4b3SWiaMzaK4spHvvX6cZx+8leLKRjYdOkdHl1N4to6//cPfYcLYLPaX1rL3dA11Le1MHj+a2xbO4ES8gZm541gYm8CbRZWcrGxk8axcPvfhOfz4rZMsv3YGADXNidNQs0YZX/rEQk5XN1HT3E5lfSvXzpzIN189jAN3Xj+TLYUVfO3uG9haGGfiuGw2HjzL1+++gaqGNrYcqWDelPGMHZ3FpHHZ3LZwBgfP1LJoZi5mcPRcPYtm5nK2toWieD1jsrKobmzleLyR/Lyp/PWL+7jjupk8fNciNh+u4Hi8gRUfvIrnd57iwWV5jMvOYl9pDeOyR5E7bjRN7Z1MGJNFe2cXHV2JdXbD7Em8fjTO+NFZlJ5v4vENhTz2uSUsXTidjk5n25EKZk0ax/YTVZRWN/PJG2YyLWcMS+ZM4p/3naG6sY3li2awMDaRyoZWKupayRplnKpuYsmcSWw7EufgmVqWLphO/gem8tN3TnPNtBxunDeZZdfOoMud3+wr51xdCx+aO5mymubE6worMDM+9+HZHCirY9akcbg7v95bxj+svJmx2VmDEY2XSPksFzO7D1jh7n8aPP8C8DF3/3KPfquAVQDXXHPNR0pKStKrWETkfWbEnOXi7mvcPd/d82Ox2GAPJyLyvpVOoJcBV3d7Pi9oExGRYZBOoL8DLDKz+WY2BrgfeCUzZYmIyECl/E9Rd+8wsy8DG4EsYK27H8xYZSIiMiDpnOWCu78KvJqhWkREJA2huFJURESSU6CLiESEAl1EJCKG9ONzzSwOpHpl0Qxg8D/dZnCEtfaw1g3hrV11D70w1P4Bd096Ic+QBno6zKygP1dKjURhrT2sdUN4a1fdQy/MtfekQy4iIhGhQBcRiYgwBfqa4S4gDWGtPax1Q3hrV91DL8y1XyI0x9BFRKRvYdpDFxGRPoQi0IfqVnf9qKPYzPab2V4zKwjappnZJjM7FnydGrSbmf1DUPN7ZnZLt+U8EPQ/ZmYPdGv/SLD8ouC1lmKda82swswOdGsb9Dp7GyMDtT9mZmXBet9rZp/tNu+RoI4jZvb73dqvuM0EHya3M2j/WfDBcpjZ2OB5UTA/b4B1X21mW83skJkdNLOH+1onI2W991F3GNb5ODPbZWb7gtr/e6rjZeo9DTt3H9EPEh/8dRxYAIwB9gFLhqmWYmBGj7a/A1YH06uBbwXTnwU2AAYsBXYG7dOAE8HXqcH01GDerqCvBa/9TIp13g7cAhwYyjp7GyMDtT8G/PUV+i4JtoexwPxgO8nqa5sBfg7cH0x/H/jzYPo/AN8Ppu8HfjbAumcDtwTTucDRoL4Rvd77qDsM69yAicH0aGBnsH4GNF4m39NwP4a9gH58024DNnZ7/gjwyDDVUszlgX4EmB1MzwaOBNM/AFb27AesBH7Qrf0HQdtsoLBb+yX9Uqg1j0tDcdDr7G2MDNT+GFcOl0u2BRKf/Hlbb9tMEACVQHbPbevCa4Pp7KCfpbH+XwY+Fab13qPuUK1zIAd4F/jYQMfL5Hsa7kcYDrnMBU53e14atA0HB14zs92WuLUewCx3Lw+mzwKzgune6u6rvfQK7ZkyFHX2NkYmfDk4NLG22yGFgdY+Hahx944r1H7xNcH82qD/gAV/yt9MYo8xNOu9R90QgnVuZllmtheoADaR2KMe6HiZfE/DKgyBPpIsd/dbgM8A/9HMbu8+0xO/rkf8aUNDUWeGx/gesBC4CSgH/neGlptxZjYR+CXwF+5e133eSF7vV6g7FOvc3Tvd/SYSd0y7Fbh+mEsaVmEI9BFzqzt3Lwu+VgC/IrEBnTOz2QDB14qge29199U+7wrtmTIUdfY2Rlrc/Vzwg9sF/JDEek+l9ipgipll92i/ZFnB/MlB/34zs9EkQvEFd38paB7x6/1KdYdlnV/g7jXAVhKHPwY6Xibf07AKQ6CPiFvdmdkEM8u9MA18GjgQ1HLhTIQHSByDJGj/4+BshqVAbfBn8Ubg02Y2Nfgz9tMkjr+VA3VmtjQ4e+GPuy0rE4aizt7GSMuFsAr8GxLr/cJ49wdnL8wHFpH4x+EVt5lg73UrcF8v6+FC7fcBW4L+/a3RgGeAw+7+dLdZI3q991Z3SNZ5zMymBNPjSRz7P5zCeJl8T8NruA/i9+dB4oyAoySOj31tmGpYQOK/3PuAgxfqIHE8bTNwDPgtMC1oN+C7Qc37gfxuy/oToCh4PNitPZ/ED85x4Duk/g+idST+TG4ncXzvoaGos7cxMlD7c0Ft75H44Zvdrf/XgjqO0O2soN62meD7uCt4Ty8CY4P2ccHzomD+ggHWvZzEoY73gL3B47Mjfb33UXcY1vmNwJ6gxgPAo6mOl6n3NNwPXSkqIhIRYTjkIiIi/aBAFxGJCAW6iEhEKNBFRCJCgS4iEhEKdBGRiFCgi4hEhAJdRCQi/h/vslNQen74CgAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD8CAYAAABn919SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAAIABJREFUeJzt3Xl8VPW9//HXZ7IRwk7CvoPggooQERStS/VSbUsXvdVbrVp76a3W29vWWq39tXa5P7292vZ6rfWi4kpVVNxaNxAoKpth3yFACAGyQMi+z3zvH3MStsxMTAaSk/t+Ph55zOTMyZxPzsy85zvfOef7NeccIiLif4H2LkBEROJDgS4i0kko0EVEOgkFuohIJ6FAFxHpJBToIiKdhAJdRKSTUKCLiHQSCnQRkU4i8VRuLD093Y0YMeJUblJExPdWrVp10DmXEWu9UxroI0aMICsr61RuUkTE98xsT0vWU5eLiEgnoUAXEekkFOgiIp2EAl1EpJNQoIuIdBIKdBGRTkKBLiLSSfgi0D/cUsBji7PbuwwRkQ7NF4G+aFshT360u73LEBHp0HwR6AEzNJm1iEh0vgh0A0LKcxGRqPwR6Gqhi4jEFDPQzWy2mRWa2cbjlt9pZlvNbJOZ/e7klRimOBcRia4lLfRngOlHLzCzy4AZwLnOubOAh+Jf2hEBMyW6iEgMMQPdObcEKD5u8feAB51ztd46hSehtiZmEFKXi4hIVK3tQx8LXGxmK8zs72Z2fqQVzWymmWWZWVZRUVGrNmaogS4iEktrAz0R6ANMAX4CzDUza25F59ws51ymcy4zIyPmhBvNMgM10EVEomttoOcB81zYSiAEpMevrGOZGU5tdBGRqFob6G8AlwGY2VggGTgYr6KOpxa6iEhsMecUNbMXgUuBdDPLA34JzAZme4cy1gE3u5N4oLhhCnQRkRhiBrpz7oYIN90Y51oiCh+1qEQXEYnGH2eKoi4XEZFYfBHoATO1z0VEYvBFoOvEIhGR2PwR6KjLRUQkFl8EOs2fsyQiIkfxRaAHvDzXELoiIpH5ItCNcKJrkgsRkcj8EehqoYuIxOSPQPcuFeciIpH5ItADXie6GugiIpH5ItAb6Vh0EZHIfBHoOmpRRCQ2fwQ66nIREYnFH4HeeJSLvhYVEYnIF4F+5MSi9q1DRKQjixnoZjbbzAq9ySyOv+3HZubM7KRNPwdHn1ikRBcRiaQlLfRngOnHLzSzocBVQG6cazrBkS4XERGJJGagO+eWAMXN3PQH4G5OYc6qgS4iElmr+tDNbAawzzm3Ls71NCugJrqISEwx5xQ9npl1BX5GuLulJevPBGYCDBs27LNuzruP8KX60EVEImtNC300MBJYZ2Y5wBBgtZkNaG5l59ws51ymcy4zIyOjVUVqLBcRkdg+cwvdObcB6Nf4uxfqmc65g3Gs6xhmjScWKdJFRCJpyWGLLwLLgHFmlmdmt538so4VUBe6iEhMMVvozrkbYtw+Im7VRGI6Dl1EJBZfnCnaNDaX8lxEJCJ/BLq6XEREYvJHoGu0RRGRmHwR6AEdhy4iEpMvAl1dLiIisfkj0NFx6CIisfgi0NF46CIiMfki0AOaVFREJCZfBHpjnOtLURGRyPwR6OpyERGJyV+B3r5liIh0aL4I9IBGWxQRickXgd4opDwXEYnIF4FupikuRERi8Uege5fqcRERiawlE1zMNrNCM9t41LL/NLOtZrbezF43s14ntcjGPvSTuREREZ9rSQv9GWD6ccvmA+Odc+cA24F741zXMTRJtIhIbDED3Tm3BCg+btkHzrkG79flhCeKPmnU5SIiEls8+tC/Dbwbh/uJSCcWiYjE1qZAN7P7gAZgTpR1ZppZlpllFRUVtXZLADj1oouIRNTqQDezW4AvAt90Uc74cc7Ncs5lOucyMzIyWrWtgFroIiIxJbbmj8xsOnA38DnnXFV8S2p2e4ACXUQkmpYctvgisAwYZ2Z5ZnYb8CjQHZhvZmvN7PGTWeSR04qU6CIikcRsoTvnbmhm8VMnoZaI9KWoiEhsvjhTVCcWiYjE5otARycWiYjE5ItA14lFIiKx+SPQNdqiiEhMvgh0HYcuIhKbLwLdvE4XTXAhIhKZPwK9qYWuRBcRicQfge5dKs5FRCLzR6Dr1H8RkZh8EujhS3W5iIhE5o9A9y4V5yIikfkj0NXlIiISk08CPXyp0RZFRCLzRaAHmsZyad86REQ6Ml8EetMUdOpzERGJqCUTXMw2s0Iz23jUsj5mNt/MdniXvU9mkUe6XEREJJKWtNCfAaYft+we4EPn3GnAh97vJ03jUS5KdBGRyGIGunNuCVB83OIZwLPe9WeBr8S5rmMcmeBCiS4iEklr+9D7O+cOeNfzgf5xqqdZjV0uodDJ3IqIiL+1+UtRF/6mMmLT2cxmmlmWmWUVFRW1ahuNoy2qfS4iEllrA73AzAYCeJeFkVZ0zs1yzmU65zIzMjJatTGd+i8iEltrA/0t4Gbv+s3Am/Epp3k6ykVEJLaWHLb4IrAMGGdmeWZ2G/AgcKWZ7QA+7/1+0piOQxcRiSkx1grOuRsi3HRFnGuJyDQFnYhITL44U1RdLiIisfkj0NFoiyIisfgi0I8MzqVEFxGJxBeBri4XEZHYfBHoGm1RRCQ2XwS6Wex1RET+r/NFoDcOzqU+dBGRyHwR6E2TRCvPRUQi8keg68QiEZGY/BHoGm1RRCQmfwS6RlsUEYnJZ4HevnWIiHRkPgl0TUEnIhKLPwLdu1QLXUQkMl8E+pFJokVEJBJfBLppcC4RkZjaFOhm9kMz22RmG83sRTPrEq/CjtmOd6k8FxGJrNWBbmaDgX8FMp1z44EE4Pp4FXbsxsIXynMRkcja2uWSCKSaWSLQFdjf9pJOZOi4RRGRWFod6M65fcBDQC5wACh1zn1w/HpmNtPMsswsq6ioqHVFNvWht7ZaEZHOry1dLr2BGcBIYBCQZmY3Hr+ec26Wcy7TOZeZkZHR2m013ldryxUR6fTa0uXyeWC3c67IOVcPzAMujE9Zx2r6UvRk3LmISCfRlkDPBaaYWVcLN6GvALbEp6xj6dR/EZHY2tKHvgJ4FVgNbPDua1ac6jqGaYILEZGYEtvyx865XwK/jFMtEWkKOhGR2Pxxpqh3qQa6iEhk/gh0jbYoIhKTLwI9oC9FRURi8kWgN54pqhOLREQi80egN43lokQXEYnEF4HeSF0uIiKR+SLQAzpuUUQkJl8EetMEF+pEFxGJyB+B7l0qzkVEIvNHoDeNttjOhYiIdGD+CHTvUke5iIhE5o9A1wQXIiIx+STQdaqoiEgsvgh0CLfSFeciIpH5J9BRA11EJJo2BbqZ9TKzV81sq5ltMbOp8SrseAEzTXAhIhJFmya4AP4LeM85d62ZJQNd41BTs9TlIiISXasD3cx6ApcAtwA45+qAuviU1cz2MHW5iIhE0ZYul5FAEfC0ma0xsyfNLC1OdZ3IdBy6iEg0bQn0RGAi8Gfn3HlAJXDP8SuZ2UwzyzKzrKKiolZvLGCoz0VEJIq2BHoekOecW+H9/irhgD+Gc26Wcy7TOZeZkZHR6o0Z+lJURCSaVge6cy4f2Gtm47xFVwCb41JVM8x02KKISDRtPcrlTmCOd4TLLuDWtpfUPPW4iIhE16ZAd86tBTLjVEtUZjrKRUQkGv+cKWqoD11EJAr/BHp7FyAi0sH5J9DNcGqhi4hE5KNA15eiIiLR+CbQNTiXiEh0vgl0DZ8rIhKdfwJdXS4iIlH5JtDRaIsiIlH5JtADho5yERGJwjeBrrFcRESi80+gYxoPXUQkCv8EulroIiJR+SbQA2Zqn4uIROGbQAcNziUiEo1vAt00ILqISFRtDnQzS/Amif5rPAqKvB3luYhINPFoof8A2BKH+4nK0GiLIiLRtCnQzWwIcA3wZHzKiSxgEFKei4hE1NYW+h+Bu4FQHGqJynSUi4hIVK0OdDP7IlDonFsVY72ZZpZlZllFRUWt3Zw32qIiXUQkkra00C8CvmxmOcBLwOVm9sLxKznnZjnnMp1zmRkZGa3fmr4UFRGJqtWB7py71zk3xDk3ArgeWOicuzFulR0noCnoRESi8s9x6OjUfxGRaBLjcSfOucXA4njcVyQay0VEJDoftdA12qKISDT+CXQdhy4iEpWPAl1T0ImIROOfQAd04KKISGT+CXR9KSoiEpVvAl0TXIiIROebQA9/KapIFxGJxD+BjrpcRESi8U2goy4XEZGofBPoGm1RRCQ63wR6QEe5iIhE5ZtAD09woUQXEYnEP4GOWugiItH4J9DV5SIiEpWPAt10HLqISBRtmVN0qJktMrPNZrbJzH4Qz8JO2B4ayUVEJJq2THDRAPzYObfazLoDq8xsvnNuc5xqO4YZuNDJuGcRkc6hLXOKHnDOrfaulwNbgMHxKux4muBCRCS6uPShm9kI4DxgRTzurzmBgCa4EBGJps2BbmbdgNeAf3POlTVz+0wzyzKzrKKiotZvB9OZoiIiUbQp0M0siXCYz3HOzWtuHefcLOdcpnMuMyMjow3b0peiIiLRtOUoFwOeArY4534fv5IiUwNdRCSytrTQLwJuAi43s7Xez9VxqusEptEWRUSiavVhi865j2mc6vMUCA/OpUgXEYnEP2eKoi4XEZFo/BPoGm1RRCQq/wQ6aqGLiETjn0A304lFIiJR+CjQ9aWoiEg0/gn09i5ARKSD80+ga4ILEZGofBPoAU1wISISlW8CXWO5iIhE559A12iLIiJR+SbQidCHvqOgnNqG4Kmvp51V1wWpqG1o7zJOmnvnbeD5ZTlxua+QD493zTtcxfq8kvYuQ3zGN4E+rE9Xcg5Vsr2gvGnZhrxSrvzDEv77w+wW38+B0urP9AYQCjle/jSXHQXl1NQH+f387eQeqvpMtX9WVXUNMT+N/MsLq/j6Y0sJtjCs6hpClFTVxaM8giH3mT4tfZJ9kN0HK1u8fmFZDS+uzGVuVl7MdY8O62355WQXHnl+BEOO+9/axBm/eI+1e1sfjtV1wWbD9fllOazcXXzC8k+yD3L5w4spra5v1facc9w+ZzU3PrmCuoaWz7sYDDnqgyd/nsaa+iBzVuzhiSW7qGymUdHcsliC3uussKym2dui2XKgjIY2/N91DSE27itt9d9Hsq+kmoZgiJr6U9fg9E2gz7x4FN1SEvmPd7c2Lfvd++HrH+2IPXHGoq2FzM3ay6X/uZgbn1zRtJOX7jzIT19dz+HKI2G3bOch/rhgO7uKKrj7tfX89LUNfG/Oan4/fzuPfLiDW55eSVlNPW+s2cfFv1vI3Ky9ADz47lbumLOakBd4B0qrI77AQiHH88tyePDdrQRDjo37StlVVEFJVR1TH1jIY4t30hAM8ebafSe0xLMLy/n79iK2FZTzzoYDAJRW1UdsiRaU1XDhgwv5hz8uobK2gdqGIH9alM2TH+1qCozm/ja7sIJVe4opq6mnuLIO5xy/eHMjZ/ziPR5bvPOYdYMhR3Zh+QlBn3uoiptnr+SbTyyntOrYgAuFHNV1Jz7ZF2wpBMIv1EgvhoKyGm58cgWf/8PfCYYcpVX1fO2xT/j875cw/pfv8/yyHN5et59nluZQ2xDi+WV7gPCb5fub8tlZVAHAqj3FbMsPvwkcqqhtdj/c98YGvvzoJyzaWti07HBlHb98axMPvLsFgIZgiKycYuqDIV5cmcuuospjwv7fXlrDHX9ZfcL+bS6I3tmQz/q8UspqGnh73X5Kq+rZuK+U4srob8i/fnsTE389n/c2HqCwrIai8loAymrq2VlU0fS/hbzHaunOg/y/NzbyxJJd3Pr0Sp5blkNhWc0xr4Wj1QdDrNpTzHeezeK+1zfy7+9s4bKHFvPhlgKe/GgXf168k+zCcib+Zj53zFl9zJtRdmEFf1qUzb6SagD2Flfx5Ee7uGPOahZuLeDZpTn89LUNfOG/PiLzt/PJLiwnFHL8eO46LnzwQ7ILK/j9B9t45MMdx9T09+1FfOG/PuLmp1fyzSeXN72hH/0mUFh+5H9an1fCjEc/5pPsgwCUVNXxzNLdfOnRj9lbXMXcrL08+dGupr99b2M+q/YUNzWymnuDXbytkCeW7KKgrKbp/7vv9Q1c9OBCbnxqBefc/wHXz1rG5v0nzP8Td3Yq+6UzMzNdVlZWq//+jwu288cFO1h816UEneOKh/8OQEpigHW/vIouSQnUB0MkJQSoD4a4/61NbC8o57uXjOb2OaupC4bompxAVV2Qm6cOZ+Lw3vzgpbUAXH32AP70TxPJOVTFl/77YypqG+jfI4Wi8lomDO3F6txwC23yyD6s2nOYS8dmsHzXIRxQVRfk6VvO5/Y5q6muD9IzNYkR6Wls3l/KlWf25/QBPXhnwwGuOXsgt182hoSA8fM3NvDC8lwAvnreYBZsLiA1OYFrJw3hscU76ZOWzI+uHMvP39jI6Iw0Hv7HCYzsm0ZRRS2PLc7mr+sOMLh3KqXV9Vw2rh9vrt3HVWf1579vmMi2/HL+sGA7l4zNIDUpgeeW5bA+L9wCyeiegnNwsCL8Yr/+/KFccUZ/fjR3LTMmDGLVnhIuGNmHgrIa3tuUT8CM/t1TCDrHD64Yy89e39D0eJwzpCc3TRnOmr0lpCYl8NTHuzlrUA9e/u5UuqWEB/L84ctr+duGA4RCjunjB/DQdefyaU4xu4oqefiDbdQ2hHj225PJLa7isnH9yMop5oF3t5JbHP4U9OMrx/KN84eSW1zFv7+zhb5pyYwb0J0XV+5tCribpgxnde5hNu0vY/pZA3hvUz7j+nfHLPzCnjS8Ny99upe+ackM7p3K+rxSkhMDXHF6P97dmE9qUgI/unIsD32wjWlj0pkwtBf9eqTw/qYCvjV1OLc9m4UBaSmJ/Oe155CcGGD3wUp+9XZ4PvTM4b3Ze7iKgrJazhnSs2lf3zRlOFed1Z/Cslp+/Mo6AF773lSycg5TWF7LUx/v5qYpw/nNV8azak8x//+drSQGjBW7ixmVnsYu71NNercUDlfVce6Qnrz2vQuprAvy7NIcrj9/KNX1QV5auZei8lpe9hoWXZMT6NElieLKOr5z8UjmrMiltLqeWy4cwSfZBzlnSC9eW51HYsBo8IKvf48UCspqSU1KIBhyXJc5hMwRvUkIBHh9dR4zLxnN2+v385cV4efsA187m3EDunPvaxvYX1JNudfo6J6S2HS9X/cUxg/uyW+/Mp575m1gyfYi+qYl88ytk7lp9gpKqupJSjAG9OxCcUUdY/p3p6Sqjj2Hqrht2khCzvH0JzmkJiXQIzWRgxV1JAaMlfd9nh5dEtm0v4z739rE+rxS6oIhEgJG37Rkpo1J56Psg7zy3amkJAW45pGPGdo7lSe+lcllDy2msi7I5af345sXDOOfn8uiX/cu5JfVcNagHmzyQve5b0+mtiHEzOezmrp6v5E5lLfX7+efLx7FD68cC8DGfaVc+/hSaupDpHdLprymgZsvHMGsJbsY068b2YUVDOrZhaBzPHPrZM4Y2KNV2Wdmq5xzmTHX81OgF3otzZHpaZRU13OwopYHvno298wLh8x1k4Ywb80+Lhzdl5TEAAu2FNK9SyLlNeEn2MxLRjFtTDoLtxby7LIcEgPGecN6c9HodP6wYDu3XDiCD7cWUF7TwHemjeShD7YD8Ld/ncanu4tJTAhw7aQh/GHBdv7n77sY2LMLz992ATfPXtn0znz24J44HNvzK0hKMCq9FujpA7qzNb+caWPS2XygLPximzaSlKQAf1q0k4B35lTIwcCeXThQeuSjZ+ObUHJCgDqvRXfHZaO5dtJQbnxyBSVVdUwc3puPdhxkeN+uHCyvpaYhdEwr5ZEbzuNn8zZQUdtAQsAY0bcrF41J5/nlewiYkRgwahtCZHRP4XBlHanJCdw0ZTjzVu+jqKKWBDPqgiHG9u/Gw9dN4EuPftzsY2QGGd1SCJhx+Rn9+MuKXG6/dHQ4DN/fRnJioKmV0/iETwgYwZAjYOH/f1RGGjMvHtX0uKYlJ1DTEGJAjy4kBIzc4iouGNmHn37hdK57fBnBkKNPWjLXTRrCvVefweyPd/Prv4bD9o/fmMBZg3pw92vrOVBSQ35ZDT/5h3G8krWXvYer+c60kSzZcZAtB8rolpLY7PcS3VMSeebbk/mXF1Y1tXqPd83ZAzl9QHcenh9+zqQkBqg9qjWX3i2Z6rpg0/Oh8X4r6hq48YLhLNpWSGlVPb3TkrlxyjBunDKcpz7azdq9Jazz3iAOVtTSLSWR9G7J5ByqYsLQXmwvKKf2qMf6sW9O5I6/rMY5mDgs3BBJS05gcO9UthdUHFPzaf268fMvnkltfZBLxmZw3ePLqA+GmDi8N3M/3dsU9o2PD4RfY9PHD+CKM/oDsHzXIa6ftZzEgHHtpCG89Olezh/RmzsvP40XV+ayZHsRIQfV9UEuHZfB4m1FJCUY3bsk8dLMKazPK+WuV9aRkhhg/g8/x7C+XfmX51fx3qZ8AL590UhmTBjEtY8vpSHkcC78Wgo51/T//PyaM7h0XD+CIccNTyynuLKOhICRnBDA4aipDz8OA3p04VBlLReflsHCrYVNz7+jnTukZ9P+BuiblsytF41oyoJGF47uS21DiJ1FFaQmJVBR20B5TUPT67xHl0SW3H0Z/7NkF189bzAj09NISmh9h0inDHSARxfu4I21+8kurOCcIT2Z+92p/PS19WzeX8aOwgomDO3FrqIKymoauO/qM7h20hCeXprD8D5d+fqkIUC4j+9nr2+gb1oK3798DL1Sk7jrlXXMW7OPHl0Sef62CxjTrxuZv11A327JfHT3ZYQnaAqrawixYEsBnxubQVpKIit3F3PXK+vo3iWRv945DTNjb3EV3VIS+d372/jc2HSmjx/I7XNW8c6GfCYN781FY9K58/IxJAaMRz7MJjkxwOkDurM69zAzJgxi9ic5/GVFLvd+4XRuuGAYb67Zx9b8cvqmJVNUUctvZownMSFAKORoCDmSEwO8sWYfr6/ZR3q3FH5wxWnUNgRpCDlKq+uZMqovv357M1vzy3jwa+fQNSWBpIQA1z2+lHOG9OKuq8bxtw0H+PrEwfRMTQLC4+fsKqqgsLyWgBlr9x7mS+cOYmDPVLbllxNyjofe30avrsnMW5PHyzOn8v6mfOZm7WX8oJ4s332Isf268+b3LyIpIcCfF2dTUlXP1NF9SU1KYPLIPtz96nrmrdnH3dPHUVMXJDU5ke9cPJKkhAB3vriGvmnJHK6qY2DPVL57ySh6dU2ioraB7l3CNX7jf5bxaU4xH/zwc4zp1w0If08y9YGFnDmwB3/712lNj92+kmpW7DrEV88bzOGqespr6hneN426hhBvrdvPecN6sb+kmkG9UlmTW0JKYoA/Lcrmga+dzXnDenOgtJoNeaWkpSTyyIc7mDYmnVEZ3RiVkdbU8iosr2HZzkOsyS3hmaU5/PyaMxg3oDvD+nQlu7CCNbklXDoug+TEAIN6pXL/W5v4YHMBKQkBZt96PueP6HPCc76qLvwm/Ku3N1NcUcf7m/MZnREOowlDe/HoP53HoYo6dh+s5CvnDebXb2+mPhjivmvO4CevrueqM/tTUx/kJ6+uZ3jfruw5VMULt13AtNPSj9lOfTBEghmBgHGoopaS6noKymoY068b81bvIzFg3DR1OCmJCU1/41w4RIf3SeMXXzqTm55awcxLRjF9/EAg3CXyo5fXUtsQYuFdn+P+tzaxcncxz357MmcN6klNfZAvP/oxN0wexq0XjQTCXZ7f/8tqvjV1BHdePoZAwHhvYz7lNfU8v3wP6/NKOXdIT26YPIyzBvXkzEE9SPBaRNsLylm0tZCLxqTz6qo8AmZ8/sx+fO+F1ZRW13PHZaO5/PR+fP3PywAY2ieVvcXVTftl7nenkl1YweYDpYzom8b4wT2ZMqovi7YVcuvTn3LLhSPokZrEC8v3MKxPVxIDxsP/eC4b9pWSW1zFLReO4Mdz1zF1dF++NXVEG5LuWC0NdJxzrf4BpgPbgGzgnljrT5o0ycVDQzDknluW4zbvL21aVlJZ555duttV1ta73EOV7t0NB1woFGrxfYZCIbfnYKUrr6lvWvZK1l733sYDLf77YDD69koq69wzn+x21XUNLbq/FbsOuYYY99kRhEIht7+kqul6XUPQOefc4cpaV3HU/mzOoYpa98aavM/0WB1tbe5h98aavGaXV9ZG3/bJVFJV5z7eUdSidUOhkKv39llLFFfUuuq6Bvdq1t4W/4/VdQ3ugXe2uPzSapdzsKLF24qHYDDUVGddQ7BFz/9IauuDrXpcdxSUuc37S10oFHKhUMi9t/GAW7ilwBWV17hHFmx3OwrK3XNLd0d8HoZCIffGmjxXWl3X6trbAshyLcjkVrfQzSwB2A5cCeQBnwI3OOc2R/qbeLTQRUT+r2lpC70tR7lMBrKdc7ucc3XAS8CMNtyfiIi0QVsCfTCw96jf87xlxzCzmWaWZWZZRUWxDy8UEZHWOenHoTvnZjnnMp1zmRkZGSd7cyIi/2e1JdD3AUOP+n2It0xERNpBWwL9U+A0MxtpZsnA9cBb8SlLREQ+q8TW/qFzrsHMvg+8DyQAs51zm+JWmYiIfCatDnQA59w7wDtxqkVERNrAN4NziYhIdKf01H8zKwL2tPLP04GDcSznVPJr7X6tG/xbu1/rBv/W7oe6hzvnYh4meEoDvS3MLKslZ0p1RH6t3a91g39r92vd4N/a/Vp3c9TlIiLSSSjQRUQ6CT8F+qz2LqAN/Fq7X+sG/9bu17rBv7X7te4T+KYPXUREovNTC11ERKLwRaCb2XQz22Zm2WZ2T3vXE42Z5ZjZBjNba2ZZ3rI+ZjbfzHZ4l73bu04AM5ttZoVmtvGoZc3WamGPeI/BejOb2MHqvt/M9nn7fa2ZXX3Ubfd6dW8zs39on6qbahlqZovMbLOZbTKzH3jLO/R+j1J3h9/vZtbFzFaa2Tqv9l95y0ea2Qqvxpe9IUwwsxTv92zv9hHtVftn1pJZMNrzh/CwAjuBUUAysA44s73rilJvDpB+3LLf4c3oBNwD/Ed71+nVcgkwEdgYq1bgauBdwIApwIoOVve/N1G0AAADFElEQVT9wF3NrHum95xJAUZ6z6WEdqx9IDDRu96d8CQxZ3b0/R6l7g6/37191827ngSs8PblXOB6b/njwPe867cDj3vXrwdebq/ny2f98UMLvTNMpDEDeNa7/izwlXaspYlzbglQfNziSLXOAJ5zYcuBXmY28NRUeqwIdUcyA3jJOVfrnNtNeLrEySetuBiccwecc6u96+XAFsLzCHTo/R6l7kg6zH739l3jbNBJ3o8DLgde9ZYfv88bH4tXgSvs6EmFOzA/BHqLJtLoQBzwgZmtMrOZ3rL+zrkD3vV8oH/7lNYikWr1w+Pwfa9bYvZR3Vodtm7vo/x5hFuMvtnvx9UNPtjvZpZgZmuBQmA+4U8MJc65hmbqa6rdu70U6HtqK24dPwS630xzzk0EvgDcYWaXHH2jC3+O88WhRX6qFfgzMBqYABwAHm7fcqIzs27Aa8C/OefKjr6tI+/3Zur2xX53zgWdcxMIz9swGTi9nUs6KfwQ6L6aSMM5t8+7LAReJ/zkKWj8mOxdFrZfhTFFqrVDPw7OuQLvRRsCnuDIx/sOV7eZJREOxTnOuXne4g6/35ur20/7HcA5VwIsAqYS7r5qHHH26Pqaavdu7wkcOsWltoofAt03E2mYWZqZdW+8DlwFbCRc783eajcDb7ZPhS0Sqda3gG95R11MAUqP6iJod8f1K3+V8H6HcN3Xe0cujAROA1ae6voaeX2xTwFbnHO/P+qmDr3fI9Xth/1uZhlm1su7ngpcSfg7gEXAtd5qx+/zxsfiWmCh96mp42vvb2Vb8kP4m/7thPu97mvveqLUOYrwN/vrgE2NtRLuf/sQ2AEsAPq0d61eXS8S/phcT7gP8bZItRI+UuBP3mOwAcjsYHU/79W1nvALcuBR69/n1b0N+EI77/NphLtT1gNrvZ+rO/p+j1J3h9/vwDnAGq/GjcAvvOWjCL/JZAOvACne8i7e79ne7aPa8znzWX50pqiISCfhhy4XERFpAQW6iEgnoUAXEekkFOgiIp2EAl1EpJNQoIuIdBIKdBGRTkKBLiLSSfwvZUs9Nci+1ZIAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.plot(range(len(tloss_list)),tloss_list)\n",
    "plt.show()\n",
    "plt.plot(range(len(vloss_list)),vloss_list)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
